### PR TITLE
Refactor assertions

### PR DESCRIPTION
**Pull request by lennartdohmann**
- Refactor assertions
- Check if a variable is set before calling
- Replace some qualifiers with imports

### CODE REVIEW

File: lib/Service/ScanService.php [hunk 1]
```diff
@@ -31,16 +32,11 @@ public function withLogger(LoggerInterface $logger): ScanService {
...
 	 */
 	public function run(): int {
 		$unscannedTagIsDisabled = $this->appConfig->getValueBool(Application::APP_ID, 'disableUnscannedTag');
-		$quantity = $this->appConfig->getValueInt(Application::APP_ID, 'scanQueueLength');
-		try {
-			$quantity = intval($quantity);
-		} catch (\Exception) {
-			$quantity = 5;
-		}
+		$quantity = $this->appConfig->getValueInt(Application::APP_ID, 'scanQueueLength', 50);
```
Comment (lennartdohmann) [inline lib/Service/ScanService.php:39]: This function can not throw and has a default value now. So the former stuff was unnecessary.

File: lib/Command/GetTagIdCommand.php [hunk 2]
```diff
@@ -40,7 +40,9 @@ protected function execute(InputInterface $input, OutputInterface $output): int
 		} catch (TagNotFoundException $e) {
 			$logger->error("Tag not found: ".$tagName." ".$e->getMessage());
 		}
-		$logger->info("tag: ".$tag->getId());
+        if (isset($tag)) {
+            $logger->info("tag: ".$tag->getId());
+        }
```
Comment (lennartdohmann) [inline lib/Command/GetTagIdCommand.php:45]: Could be undefined on calling

File: tests/unittests/ScanServiceTest.php [hunk 3]
```diff
@@ -109,7 +115,12 @@ public function testRun_unscannedTagDisabled_unscannedTagShouldNotBeCreated() {
 		$scanService->run();
 	}
 
-	public function testRun_unscannedTagEnabled_unscannedTagShouldBeCreated() {
+    /**
+     * @return void
+     * @throws Exception
+     * @throws \PHPUnit\Framework\MockObject\Exception
+     */
+    public function testRun_unscannedTagEnabled_unscannedTagShouldBeCreated() {
```
Comment (unglaublicherdude) [inline tests/unittests/ScanServiceTest.php:123]:
```suggestion
    public function testRun_unscannedTagEnabled_unscannedTagShouldBeCreated(): void {
```

File: tests/unittests/ScanServiceTest.php [hunk 4]
```diff
@@ -24,7 +25,12 @@ public function setUp(): void {
 		$this->logger = new TestLogger();
 	}
 
-	public function testRun_unscannedTagDisabled_unscannedTagShouldNotBeCreated() {
+    /**
+     * @return void
+     * @throws Exception
+     * @throws \PHPUnit\Framework\MockObject\Exception
+     */
+    public function testRun_unscannedTagDisabled_unscannedTagShouldNotBeCreated() {
```
Comment (unglaublicherdude) [inline tests/unittests/ScanServiceTest.php:33]:
```suggestion
    public function testRun_unscannedTagDisabled_unscannedTagShouldNotBeCreated(): void {
```

File: tests/unittests/TagServiceTest.php [hunk 5]
```diff
@@ -7,23 +7,32 @@
...
 		});
 		return $tagManager;
 	}
 
-	public function testSetWontScan_ShouldDeleteCleanAndMaliciousAndAssignWontScan() {
+    /**
+     * @return void
+     * @throws Exception
+     */
+    public function testSetWontScan_ShouldDeleteCleanAndMaliciousAndAssignWontScan() {
```
Comment (unglaublicherdude) [inline tests/unittests/TagServiceTest.php:35]:
```suggestion
    public function testSetWontScan_ShouldDeleteCleanAndMaliciousAndAssignWontScan(): void {
```

File: tests/unittests/TagServiceTest.php [hunk 6]
```diff
@@ -52,7 +61,11 @@ public function testSetWontScan_ShouldDeleteCleanAndMaliciousAndAssignWontScan()
 		$tagService->setTag(self::$OBJECT_ID_1, TagService::WONT_SCAN);
 	}
 
-	public function testSetWontScan_TagAlreadySet_ShouldNotDoAnything() {
+    /**
+     * @return void
+     * @throws Exception
+     */
+    public function testSetWontScan_TagAlreadySet_ShouldNotDoAnything() {
```
Comment (unglaublicherdude) [inline tests/unittests/TagServiceTest.php:68]:
```suggestion
    public function testSetWontScan_TagAlreadySet_ShouldNotDoAnything(): void {
```

File: tests/unittests/TagServiceTest.php [hunk 7]
```diff
@@ -66,7 +79,11 @@ public function testSetWontScan_TagAlreadySet_ShouldNotDoAnything() {
 		$tagService->setTag(self::$OBJECT_ID_1, TagService::WONT_SCAN);
 	}
 
-	public function testSetWontScan_UnscannedTagDoesNotExist_SouldTagFileWithWontScan_WithNoException() {
+    /**
+     * @return void
+     * @throws Exception
+     */
+    public function testSetWontScan_UnscannedTagDoesNotExist_ShouldTagFileWithWontScan_WithNoException() {
```
Comment (unglaublicherdude) [inline tests/unittests/TagServiceTest.php:86]:
```suggestion
    public function testSetWontScan_UnscannedTagDoesNotExist_ShouldTagFileWithWontScan_WithNoException(): void {
```

File: tests/unittests/TagServiceTest.php [hunk 8]
```diff
@@ -80,7 +97,11 @@ public function testSetWontScan_UnscannedTagDoesNotExist_SouldTagFileWithWontSca
 		$tagService->setTag(self::$OBJECT_ID_1, TagService::WONT_SCAN);
 	}
 
-	public function testSetWontScan_NoneVaasTagIsSet_SouldTagFileWithWontScan_AndNotDeleteTheNoneVaasTag() {
+    /**
+     * @return void
+     * @throws Exception
+     */
+    public function testSetWontScan_NoneVaasTagIsSet_ShouldTagFileWithWontScan_AndNotDeleteTheNoneVaasTag() {
```
Comment (unglaublicherdude) [inline tests/unittests/TagServiceTest.php:104]:
```suggestion
    public function testSetWontScan_NoneVaasTagIsSet_ShouldTagFileWithWontScan_AndNotDeleteTheNoneVaasTag(): void {
```

File: lib/Service/ScanService.php (hunk 1 above)
Comment (unglaublicherdude) [reply lib/Service/ScanService.php:39]: so you sneaked in some changes here? :smile:

File: lib/Service/TagService.php [hunk 9]
```diff
@@ -82,11 +83,11 @@ public function setTag(int $fileId, string $tagName): void {
 
 		if (isset($filesTagIds[$fileId])) {
 			foreach ($filesTagIds[$fileId] as $tagId) {
-				if ($tagId != $tag->getId() && \in_array($tagId, $vaasTagIds)) {
```
Comment (unglaublicherdude) [inline lib/Service/TagService.php:85]: has this change any value?

The slash is making it clear to use the function from the [global namespace](https://www.php.net/manual/en/language.namespaces.global.php).

File: lib/Service/TagService.php [hunk 10]
```diff
@@ -82,11 +83,11 @@ public function setTag(int $fileId, string $tagName): void {
...
+				if ($tagId != $tag->getId() && in_array($tagId, $vaasTagIds)) {
 					$this->tagMapper->unassignTags(strval($fileId), 'files', [$tagId]);
 				}
 			}
-			if (\in_array($tag->getId(), $filesTagIds[$fileId])) {
```
Comment (unglaublicherdude) [inline lib/Service/TagService.php:89]: same

File: tests/unittests/TagServiceTest.php [hunk 11]
```diff
@@ -7,23 +7,32 @@
 use OCP\SystemTag\ISystemTag;
 use OCP\SystemTag\ISystemTagManager;
 use OCP\SystemTag\ISystemTagObjectMapper;
+use PHPUnit\Framework\MockObject\Exception;
```
Comment (unglaublicherdude) [inline tests/unittests/TagServiceTest.php:10]: This for example is where I really don't like the doc comments. The import is now just for the comment? It is not used in any code-part and the functions here are test functions, that will never be called by any developer from the outside.

Also the php autocompleters will already show you the signature of the function, so you will see the informations already in your IDE.
image

The doc-comment is just adding the throws but they are not even complete.
image

The InvalidArgumentException exception for example is from the global namespace.

**Comment by unglaublicherdude** [comment]
Thank you for the PR. I like most of the changes but also made suggestions.

**Review by unglaublicherdude — CHANGES_REQUESTED** [review]

File: lib/Command/GetTagIdCommand.php (hunk 2 above)
Comment (unglaublicherdude) [reply lib/Command/GetTagIdCommand.php:45]: yeah, but its better to return 1 in the catch, so its an actual shell error. Did adjust that in my pr

**Comment by unglaublicherdude** [comment]
I rebased this branch

**Review by pstadermann — APPROVED** [review]

**Pull request merged**